### ISSUE TITLE
'Zero-argument form of "super" call is valid only within a method' false positive when creating subclass in method

### ISSUE THREAD

**Issue by ThiefMaster** [microsoft/pyright#6093]
### Environment data

-   Language Server version: 2023.9.30
-   OS and version: linux x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.12
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: off

### Code Snippet

```python
class Foo(Bar):
    def create_schema(self):
        class CustomSchema(super().create_schema()):
            pass
        return CustomSchema
```

### Output

> Zero-argument form of "super" call is valid only within a method

However, this usage is perfectly valid. I guess pylance already things it's in the class scope - but the superclass is resolved within the containing method's scope, hence `super()` is valid there. In fact, this is based on real code in my application which works fine.

**Comment by erictraut** [comment]
Could someone from the pylance team please transfer this to the pyright repo, since it's a core type checker issue? Thanks!

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.331, which I just published. It will also be included in a future release of pylance.
